python_modules/dagster/dagster/_utils/typing_api.py

```diff
@@ -128,3 +128,17 @@ def is_typing_type(ttype):
         or ttype is typing.Dict
         or ttype is typing.List
     )
+
+
+def flatten_annotation_unions(annotation: typing.Type) -> typing.Iterable[typing.Type]:
```

Comment:
Function makes sense, but the name was def counterintuitive for whatever reason. Had to read docblock and code to understand what was going on here.

Code like this also always spooks me in terms of python version compatibility, so it is worth writing a bunch of unit tests and making sure it passes all of them when we land it in master and run the full suite.

Reply:
Also in the example in the doc block would it not return `(str, int, float, None)` because of the optional?

Reply:
> Function makes sense, but the name was def counterintuitive for whatever reason. Had to read docblock and code to understand what was going on here.

Yeah I had trouble coming up with a good name. Not sure if this helps, but I switched it to just `flatten_unions`.

> Also in the example in the doc block would it not return (str, int, float, None) because of the optional?

You are right - thanks for catching. Fixed this.

> Code like this also always spooks me in terms of python version compatibility, so it is worth writing a bunch of unit tests and making sure it passes all of them when we land it in master and run the full suite.

👍 👍  added.